plugin/ochttp/httptrace.go

```diff
@@ -132,34 +132,39 @@ func (t *traceTransport) CancelRequest(req *http.Request) {
 	}
 }
 
-// NewHandler returns a http.Handler from the given handler
-// that is aware of the incoming request's span.
-// The span can be extracted from the incoming request in handler
-// functions from incoming request's context:
+// NewHandler returns a new handler.
+func NewHandler(base http.HandlerFunc) *Handler {
```

Comment:
Am afraid changing this signature from http.Handler to http.HandlerFunc will restrict the flexibility and interop with the outside world. It is easier to create an http.Handler than an http.HandlerFunc.
Let's make this
```go
func NewHandler(base http.Handler) *Handler
```
even the function suffix says "Handler" :)

Reply:
@odeke-em agreed if this is the recommended way to create a Handler then it's odd that it only takes a HandlerFunc not Handler.

Reply:
Done